reference/lib/ReferenceOrderValidator.sol

```diff
@@ -238,7 +263,30 @@ contract ReferenceOrderValidator is
         }
 
         // Return order hash, new numerator and denominator.
-        return (orderHash, numerator, denominator);
+        return (orderHash, uint120(numerator), uint120(denominator));
+    }
+
+    /**
+     * @dev Internal function to derive the greatest common divisor of two
+     *      values using the classical euclidian algorithm.
+     *
+     * @param a The first value.
+     * @param b The second value.
+     *
+     * @return greatestCommonDivisor The greatest common divisor.
+     */
+    function _greatestCommonDivisor(uint256 a, uint256 b)
+        internal
+        pure
+        returns (uint256 greatestCommonDivisor)
+    {
+        while (b > 0) {
+            uint256 c = b;
+            b = a % c;
```

Comment:
This will be checked against `c != 0`, but `c` can't be 0. Potentially use unchecked block for the mod.

Reply:
We've got a rule of no unchecked and no assembly in the reference implementation!